### PR TITLE
feat(extra/natives): Add many vehicle natives

### PR DESCRIPTION
**Pull request by NoxieNya**
also fixes some existing natives

Thanks to [ikt](https://github.com/E66666666/GTAVManualTransmission/blob/master/Gears/Memory/VehicleExtensions.cpp) for the offsets

### CODE REVIEW

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 1]
```diff
@@ -371,9 +381,37 @@ static HookFunction initFunction([]()
 		auto location = hook::get_pattern<char>("F6 83 ? ? ? ? 07 75 ? 44 0F");
 
 		CurrentRPMOffset = *(uint32_t*)(location - 42);
+		ClutchOffset = CurrentRPMOffset + 12;
 		ThrottleOffsetOffset = CurrentRPMOffset + 16;
 	}
 
+	{
+		auto location = hook::get_pattern<char>("45 0f 57 ? f3 0f 11 ? ? ? 00 00 f3 0f 5c");
```
Comment (Disquse) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:389]: Can you please uppercase all pattern bytes?

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 2]
```diff
@@ -371,9 +381,37 @@ static HookFunction initFunction([]()
...
+
+		WheelSuspensionCompressionOffset = *(uint32_t*)(location + 8);
+		WheelRotationSpeedOffset = *(uint32_t*)(location + 8) + 0xC;
+	}
+
+	{
+		char* location;
+		if (Is2060()) {
+			location = hook::get_pattern<char>("0F 2F ? ? ? 00 00 0F 97 C0 EB ? D1");
```
Comment (Disquse) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:398]: What about `0F 2F 8B ? ? 00 00 0F 92 C0`? Seems valid for 1604 *and* 2060.

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp (hunk 1 above)
Comment (NoxieNya) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp:389]: My mistake, didn't even notice. Will fix.

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp (hunk 2 above)
Comment (NoxieNya) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp:398]: I noticed that, but ikt had the two separated in his code so I figured he would have had a reason for doing that. He certainly knows a lot more than I do.

Comment (NoxieNya) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp:398]: I tested the reduced pattern and it doesn't work. That extra byte is necessary.

File: code/components/extra-natives-five/src/VehicleExtraNatives.cpp [hunk 3]
```diff
@@ -371,9 +381,37 @@ static HookFunction initFunction([]()
...
+			location = hook::get_pattern<char>("0F 2F ? ? ? 00 00 0F 97 C0 EB DA");
+		}
+		WheelSteeringAngleOffset = (*(uint32_t*)(location + 3));
+		WheelBrakePressureOffset = (*(uint32_t*)(location + 3)) + 0x4;
+		WheelPowerOffset = (*(uint32_t*)(location + 3)) + 0x8;
+		WheelTractionVectorLengthOffset = (*(uint32_t*)(location + 3)) - 0x14;
+	}
+
+	{
+		auto location = hook::get_pattern<char>("75 11 48 8b 01 8b 88");
```
Comment (Disquse) [inline code/components/extra-natives-five/src/VehicleExtraNatives.cpp:410]: Wrong capitalization here too

Comment (NoxieNya) [reply code/components/extra-natives-five/src/VehicleExtraNatives.cpp]: I guess I'm blind. Sorry, fixed now.

**Comment by blattersturm** [comment]
@Disquse can this be considered good to go?

**Comment by NoxieNya** [comment]
Sorry, just realized I smoothbrained the last commit and had it greater than 2189 when it should've remained 2060.

**Review by Disquse — APPROVED** [review]
LGTM. Tested across all game builds, works good. Thanks!

**Pull request merged**